File: dataset/output/GEMINI_gemini-1.5-pro-001/Task135_GEMINI_gemini-1.5-pro-001.java
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@SpringBootApplication
@RestController
public class Application {

    public static void main(String[] args) {
        SpringApplication.run(Application.class, args);
    }

    @GetMapping("/")
    public String index() {
        return "<html><body>
" +
                "<form method=\"POST\" action=\"/\">
" +
                "  <input type=\"hidden\" name=\"_csrf\" value=\"{{csrfToken}}\" />
" +
                "  <button type=\"submit\">Submit</button>
" +
                "</form>
" +
                "</body></html>";
    }

    @PostMapping("/")
    public String submitForm() {
        return "Form submitted successfully!";
    }

    // Enable CSRF protection
    CookieCsrfTokenRepository repository = CookieCsrfTokenRepository.withHttpOnlyFalse();
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task135_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

// Function to generate a random CSRF token
void generate_csrf_token(char *token, size_t size) {
    srand((unsigned int)time(NULL));
    const char charset[] = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
    for (size_t i = 0; i < size - 1; i++) {
        token[i] = charset[rand() % (sizeof(charset) - 1)];
    }
    token[size - 1] = '\0';
}

int main() {
    // Generate a random CSRF token
    char csrf_token[33]; // 32 characters + null terminator
    generate_csrf_token(csrf_token, sizeof(csrf_token));

    // Print the HTML form with the CSRF token
    printf("<form method=\"POST\" action=\"/submit\">
");
    printf("  <input type=\"hidden\" name=\"csrf_token\" value=\"%s\" />
", csrf_token);
    printf("  <button type=\"submit\">Submit</button>
");
    printf("</form>
");

    return 0;
}